e2e_test/ethersjs-api-check/test-precompile-wrappers/cip20.test.ts

```diff
@@ -0,0 +1,33 @@
+import {ethers} from 'ethers';
+import {assert} from 'chai';
+import 'mocha';
+import * as fs from 'fs';
+
+
+describe("CIP20", function () {
+  let cip20: any;
+  this.timeout(25000);
+
+  before(async () => {
+    const provider = new ethers.providers.JsonRpcProvider(process.env.npm_config_networkaddr);
+    const signerKey: any = ethers.utils.hexlify(process.env.npm_config_signerkey!)
+    const signer = new ethers.Wallet(signerKey, provider);
+
+    const contractJSON = fs.readFileSync("../../compiled-system-contracts/Cip20Test.json", 'utf8');
+    const CIP20TestFactory = ethers.ContractFactory.fromSolidity(contractJSON, signer);
+    cip20 = await CIP20TestFactory.deploy();
+  });
+
+  it("should run hashes", async () => {
+    const vectors = require("./cip20.json")
+    for (const vector of vectors) {
+      let preimage = `0x${vector.preimage}`;
```

Comment:
were you able to verify the unit test data is correct?

Reply:
I only tested one or two of the hashes against the CLI implementation on my machine. The test data was already included in the [original repository](https://github.com/prestwich/b12-sol), so I did not thoroughly check it again.

Reply:
I tested all hashes except blake2 against https://emn178.github.io/online-tools/sha3_512.html for the first test case and the results matched.